src/string_object.cpp

```diff
@@ -138,6 +138,58 @@ Value StringObject::center(Env *env, Value length, Value padstr) {
     return new StringObject { result, m_encoding };
 }
 
+Value StringObject::chomp(Env *env, Value record_separator) {
+    const char *cstr = m_string.c_str();
+    size_t end_idx = m_string.length();
+
+    if (record_separator.is_null()) {
+        if (cstr[end_idx-1] == '\r') {
+            --end_idx;
+        } else if (cstr[end_idx-1] == '\n') {
+            if (cstr[end_idx-2] == '\r') {
+                --end_idx;
+            }
+            --end_idx;
+        }
+        
+        return new StringObject(m_string.substring(0, end_idx));
+    } 
+
+    record_separator->assert_type(env, Object::Type::String, "String");
+    StringObject *sep = record_separator->as_string();
+    const char *csep = sep->m_string.c_str();
+    size_t sep_len = sep->m_string.length();
+
+    if (sep_len == 0) {
+        while (end_idx > 0 && cstr[end_idx - 1] == '\n') {
+            if (end_idx > 1 && cstr[end_idx-2] == '\r') {
+                --end_idx;
+            }
+            --end_idx;
+        }
```

Comment:
I noticed this block of code is similar to the one starting on 145, but not quite the same. If I'm reading it correctly, passing an empty string as the separator is the same as passing no record separator. There shouldn't be a reason the code has to repeat, so we can share the same code if we rearrange things. Using a lambda to share code is fine, or checking for a null/empty separator up top is fine too.

```sh
→ cat > wat.rb
s = "foo\n\r"
p s.chomp("")

→ ruby wat.rb
"foo\n\r"

→ bin/natalie wat.rb
"foo\n\u000D"
```

Reply:
There are actually two differences between passing an empty string and  passing no record separator. With an empty string, \r by itself is not removed unless \n comes after. Also, with an empty string, all of the trailing separators are removed instead of just the last. 
i.e. `"hello\r\n\r\n".chomp => "hello\r\n"; "hello\r\n\r\n".chomp('') => "hello"`

Reply:
I'm glad the null character didn't produce any unexpected results! To be honest I hadn't thought of that, and was working with c strings only because I was unfamiliar with TM::String, but in the future I will be sure to use TM::String wherever I can.

Reply:
OK, very good. Thanks for explaining that.

Where my confusion came in was seeing different output in MRI vs Natalie, but that's just a separate bug we can fix later, i.e. `"\r"` is printed as `"\u000D"`.